behave/features/steps/specialist_provider.py

```diff
@@ -65,65 +71,149 @@ def step_impl(context):
-    accepted_tab.click()  
-    
-    
+    accepted_tab.click()
+
+
 @then(u'I can see my accepted case reference number')
 def step_impl(context):
     # context.selected_case_ref
     # click on the 'back to cases' link
     my_case = context.helperfunc.find_by_xpath(f"//*[text()='{context.selected_case_ref}']")
     assert my_case is not None
-     
 
-    
+
+@given(u'that I am viewing a case that I have accepted as a specialist provider')
+def step_impl(context):
+    case_reference = CLA_SPECIALIST_CASE_TO_ACCEPT
+    login_url = f"{CLA_FRONTEND_URL}/provider/{case_reference}/diagnosis/"
+    context.helperfunc.open(login_url)
+
+
+@given(u'I select the Legal help form')
+def step_impl(context):
+    wrapper = context.helperfunc.find_by_css_selector('.CaseBar-actions')
+    legal_help_form_link = wrapper.find_element_by_xpath("//a[text()='Legal help form']")
+    assert legal_help_form_link is not None, "Could not find legal help form link"
+    legal_help_form_link.click()
+
+
+def assert_your_details(table, root_element):
+    for row in table:
+        label_element = root_element.find_element_by_xpath(f".//*[text()='{row['field']}']")
+        assert label_element is not None, f"Could not find question on legal help form: {row['field']}"
+        parent_element = label_element.find_element_by_xpath(f"./..")
+        try:
+            value_element = parent_element.find_element_by_tag_name('input')
+        except NoSuchElementException:
+            value_element = parent_element.find_element_by_tag_name('textarea')
+
+        assert value_element is not None, f"Could not find value for question: {row['field']}"
+
+        expected_label = row['field'].upper()
+        expected_value = row['value'].upper()
+        actual_value = value_element.get_attribute('value').upper()
+        actual_label = label_element.text.upper()
+
+        assert actual_label == expected_label, f"Expected label: {expected_label} - Actual value:{actual_label}"
+        assert actual_value == expected_value, f"Expected value: {expected_value} - Actual value:{actual_value}"
+
+
+def assert_four_column_table(table, root_element):
+
+    def assert_cell(element, question, expected_value):
+        value = element.get_attribute("value")
+        assert value == expected_value, f"Question: {question} - Expected: {expected_value} - Actual: {value}"
+
+    for row in table:
+        question = row[0]
+        label_element = root_element.find_element_by_xpath(f".//*[text()='{question}']")
+        assert label_element is not None, f"Could not find question on legal help form: {question}"
+        parent_element = label_element.find_element_by_xpath("./..//ancestor::tr")
+        elements = parent_element.find_elements_by_tag_name("td input")
+
+        assert_cell(elements[0], question, row.cells[1])
+        if len(row) > 2 and row[2].lower() != 'n/a':
+            assert_cell(elements[1], question, row[2])
+        if len(row) > 3 and row[2].lower() != 'n/a':
```

Comment:
should this be row[3].lower() that is compared? If so this can be simplified to a loop.

Reply:
Nice catch! updated